cpp/orbbec/record.cpp

```diff
@@ -102,10 +118,14 @@ int main(int argc, char *argv[]) {
     int gain = -1;
     int brightness = -1;
     bool print = false;
+    bool autoSubfolders = false;
+
     for (size_t i = 1; i < arguments.size(); ++i) {
         const std::string &argument = arguments.at(i);
         if (argument == "--output")
             config.recordingFolder = arguments.at(++i);
+        else if (argument == "--auto_subfolders")
+            autoSubfolders = true;
```

Comment:
I think the automatic creation of timestamped folders is the most commonly desired mode of operation, so it's a bit awkward that you need to set this parameter to get that behavior(. Instead, the argument could be removed by making it work like this:
* If `recordingFolder` is set, record to that directory without any subfolders (overwrites if run twice). This allows the user to save data exactly where he wants, as he can create the parent folder himself.
* If `recordingFolder` is not set, save in `data/%Y-%m-%d_%H-%M-%S`. It probably does not matter much that the user can't set name of the parent folder, but you could also add an optional argument to set that path.

Reply:
Now it works like this:
If `--output` is set, then recordings are saved to `output`
If `--output` is not set, then recordings are saved to `data/timestamp` (default)
If `--output --auto_subfolders` is set, then recordings are saved to `output/timestamp`